src/wallet/wallet.cpp

```diff
@@ -3618,14 +3714,41 @@ CWallet* CWallet::CreateWalletFromFile(const std::string walletFile)
         walletInstance->SetMaxVersion(nMaxVersion);
     }
 
+
+    std::string hdWatchOnly = GetArg("-hdwatchonly", "");
+    CExtPubKey extPubKey;
+    if (!hdWatchOnly.empty()) {
+        CBitcoinExtPubKey bitcoinExtPubKey;
+        auto extpubKeyPrefix = chainparams.Base58Prefix(CChainParams::Base58Type::EXT_PUBLIC_KEY);
+        if (!bitcoinExtPubKey.SetString(hdWatchOnly.c_str(), extpubKeyPrefix.size())) {
+            InitError(_("Invalid ExtPubKey format"));
+            return NULL;
+        }
+        extPubKey = bitcoinExtPubKey.GetKey();
+    }
+
+    if (!fFirstRun && !hdWatchOnly.empty()) {
```

Comment:
this is normally checked. Actually I am testing it on https://github.com/bitcoin/bitcoin/pull/9728/files#diff-08fcd10ff4d7d4f9f0249ba978b3b4d4R49 this is strange...